File: backend/src/main/java/com/pharma/backend/entity/ChiTietGioHang.java
package com.pharma.backend.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "chi_tiet_gio_hang")
public class ChiTietGioHang {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ma_chi_tiet_gio_hang")
    private Long maChiTietGioHang;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ma_gio_hang", nullable = false)
    private GioHang gioHang;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ma_san_pham", nullable = false)
    private SanPham sanPham;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ma_don_vi_san_pham", nullable = false)
    private DonViSanPham donViSanPham;

    @Column(name = "so_luong", nullable = false)
    private Integer soLuong;

    @Column(name = "don_gia", nullable = false)
    private BigDecimal donGia;

    @Column(name = "thanh_tien", nullable = false)
    private BigDecimal thanhTien;
}

File: frontend/src/features/gio-hang/types/ChiTietGioHang.ts
import type { DonViBanSanPham } from "../../../shared/types/DonViBanSanPham";

export interface ChiTietGioHang {
  maChiTietGioHang: number;

  maSanPham: number;
  tenSanPham: string;
  hinhAnh: string | null;

  maDonViSanPham: number;
  maDonViTinh: number | null;
  tenDonViTinh: string | null;
  kyHieuDonViTinh: string | null;

  soLuong: number;
  donGia: number;
  thanhTien: number;

  danhSachDonViBan: DonViBanSanPham[];
}